掲示板新規登録機能の実装が完了いたしました。

YukiIshizaki0525 opened the pull request:
■変更点
・投稿内容はcontentというカラムで管理
・未入力で登録不可
・投稿内容は140字以内
・_error_messages.html.erbをtemplateフォルダからsharedフォルダに移動
・投稿時のフラッシュメッセージを設定
・エラーメッセージ追加のため、rails-i18n Gemの追加
・ja.ymlを作成し、contentを「投稿内容」としてエラーメッセージの際、日本語化するようにしました。
・投稿内容に不備がある場合は、登録フォームにリダイレクトし、入力した内容が残っている仕様にしました。
・未ログインの状態で投稿フォームのパスを入力しても、トップページに戻され、「ログインまたは新規登録が必要です」というフラッシュメッセージが表示される仕様にしました。

YukiIshizaki0525 committed:
掲示板新規登録機能の実装が完了いたしました。

masato-sa93 commented:
繰り返しになりますが、コミットは細かく分けましょうー！
1コミット=1つの意味、にするのが理想です。

またコミットメッセージは何をしたか書いておくとレビュワーが見やすいです！

それと、チーム間でのレビュー結果はPR上で行ってください！
Slackだと流れてしまうので…

masato-sa93 commented on app/controllers/boards_controller.rb:
```diff
@@ -0,0 +1,31 @@
+class BoardsController < ApplicationController
+
+  def new
+    if user_signed_in?
```
deviseを使用してるので、`authenticate_user!`の記述で対応しましょう！

YukiIshizaki0525 replied:
修正いたしました。

masato-sa93 commented on app/controllers/boards_controller.rb:
```diff
@@ -0,0 +1,31 @@
+      flash[:notice] = "「#{board.title}」の掲示板を作成しました"
+      redirect_to board
+    else
+      redirect_to new_board_path, flash: {
```
redirectではなくrenderを使用しましょう！

masato-sa93 commented on app/models/board.rb:
```diff
@@ -0,0 +1,3 @@
+class Board < ApplicationRecord
+  validates :content, presence: true, length: { maximum: 140 }
```
userモデルとのアソシエーションを記述しましょう！

YukiIshizaki0525 replied:
修正いたしました。

masato-sa93 commented on app/controllers/boards_controller.rb:
```diff
@@ -0,0 +1,31 @@
+  private
+
+    def board_params
+      params.require(:board).permit(:content)
```
これだとUserとBoardが紐づいていないので、修正しましょう！

YukiIshizaki0525 replied:
修正しました。

masato-sa93 commented on db/migrate/20210307095420_create_boards.rb:
```diff
@@ -0,0 +1,9 @@
+  def change
+    create_table :boards do |t|
+      t.text :content
+
```
usersテーブルへの外部キーを追加しましょう！

masato-sa93 reviewed (changes requested):
レビュー完了しました
ご確認＆修正お願いしますー！

YukiIshizaki0525 committed:
BoardsとUserモデルの紐付けしました
Boardsテーブルにuser_idとして外部キーを追加
redirect_toからrenderに変更
user_signed_in?からauthenticatior_user!に変更

masato-sa93 commented on app/controllers/boards_controller.rb:
```diff
@@ -1,31 +1,24 @@
-      flash[:notice] = "「#{board.title}」の掲示板を作成しました"
-      redirect_to board
+      flash[:notice] = "投稿が完了しました。"
+      render :new
```
投稿に成功した場合にnewにrenderさせてしまうと投稿した内容が確認できずユーザビリティが落ちるので、showかindexにredirectさせてあげるのがいいと思いますー！

masato-sa93 commented on app/controllers/boards_controller.rb:
```diff
@@ -1,31 +1,24 @@
 
     def board_params
-      params.require(:board).permit(:content)
+      params.permit(:content, :user_id)
```
requireなくてcontent取れてますかね、、、？取れてるなら問題ないですー！

masato-sa93 commented on app/controllers/boards_controller.rb:
```diff
@@ -1,31 +1,24 @@
 
   def create
-    board = Board.new(board_params)
+    board = current_user.boards.new(board_params)
```
hidden_fieldでuser_idをpostしてるならcurrent_userはなくてもいいですね。
どちらかでいいです。

masato-sa93 reviewed (approved):
レビュー期限過ぎてるのでApproveしてしまいますー！
修正したらそのままマージしてください。

YukiIshizaki0525 committed:
hidden_fieldの記述を削除しcurrent_userにしました。
ストロングパラメーターのrequireの記述を削除しました。
投稿後、投稿一覧ページに修正する仕様に変更
